lib/src/booleans/boolean_point_on_line.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:turf/helpers.dart';
+
+/// Returns true if a point is on a line. Accepts an optional parameter to ignore the
+/// start and end vertices of the [Linestring].
+/// The [ignoreEndVertices=false] controls whether to ignore the start and end vertices.
+/// [epsilon] is the Fractional number to compare with the cross product result.
+/// It's useful for dealing with floating points such as lng/lat points
+/// example:
+/// ```dart
+/// var pt = Point(coordinates:Position.of([0, 0]));
+/// var line = LineString(coordinates: [
+///   Position.of([-1, -1]),
+///   Position.of([1, 1]),
+///   Position.of([1.5, 2.2]),
+/// ]);
+/// var isPointOnLine = booleanPointOnLine(pt, line);
+/// //=true
+/// ```
+bool booleanPointOnLine(Point pt, LineString line,
+    {bool ignoreEndVertices = false, num? epsilon}) {
+  for (var i = 0; i < line.coordinates.length - 1; i++) {
+    dynamic ignoreBoundary = false;
+    if (ignoreEndVertices) {
+      if (i == 0) {
+        ignoreBoundary = "start";
+      }
+      if (i == line.coordinates.length - 2) {
+        ignoreBoundary = "end";
+      }
+      if (i == 0 && i + 1 == line.coordinates.length - 1) {
+        ignoreBoundary = "both";
```

Comment:
let's convert the ignoreBoundary to an enum, something like this:
```dart
enum BoundaryType {
    none,
    start,
    end,
    both
}
```

That way, we have a little more "safety"